cmd/controller/main.go

```diff
@@ -192,6 +199,8 @@ func main() {
 	flag.BoolVar(&profiling, "enable-profiling", false, "when set, enables the endpoints on the http server under /debug/pprof/")
 	flag.BoolVar(&externalCloudProvider, "external-cloud-provider", false, "when set, kubelets will receive --cloud-provider=external flag")
 	flag.DurationVar(&skipEvictionAfter, "skip-eviction-after", 2*time.Hour, "Skips the eviction if a machine is not gone after the specified duration.")
+	flag.StringVar(&nodeHttpProxy, "node-http-proxy", "", "If set, this proxy will be configured on all nodes.")
+	flag.StringVar(&nodeImageRegistry, "node-image-registry", "", "If set, this image registry will be used for pulling all required images on the node.")
```

Comment:
Can we rename this to `insecure-registry`? That is what it is after all, it doesn't change any behaviour about what gets pulled from which location, it just adds an insecure registry. Also we might want this to be a string slice and not a string

Reply:
If set, it changes the source for the pause & hyperkube container.

Reply:
Similiarily to hyperkube container, pause container should be a dedicated option, probably a flag?

Right now we make the assumptions that:

* User has a registry and its insecure
* User puts hyperkube image into that registry instead of pulling it from the internet (which would be possible, if we did set the proxy flags for docker)
* User puts pause image into that registry instead of pulling it from the internet (which again could be done by setting up the proxy config for Docker)

Reply:
Done with https://github.com/kubermatic/machine-controller/pull/559/commits/649786bd2d98d7c54b81975417e047a9a654ab42